### PR TITLE
1351 debug print category

### LINKED ISSUE #1351: Split debug prints into terse, normal, verbose
Do it.

### PR DESCRIPTION
**Pull request by lifflander**
Fixes #1351

- [x] make interface changes for debug print
- [x] add new user command-line options to control terse, normal, verbose
- [x] disable verbose at compile-time with release?
- [x] conform all calls to the new interface
  - [x] barrier.cc compiles
  - [x] ...

### CODE REVIEW

File: src/vt/configs/arguments/args.cc [hunk 1]
```diff
@@ -619,6 +620,16 @@ void ArgConfig::postParseTransform() {
...
   config_.vt_trace_pmpi = contains(arg_trace_mpi, "external");
+
+  if (config_.vt_debug_level == "terse" or config_.vt_debug_level == "0") {
+    config_.vt_debug_level_val = 2;
+  } else if (config_.vt_debug_level == "normal" or config_.vt_debug_level == "1") {
+    config_.vt_debug_level_val = 4;
+  } else if (config_.vt_debug_level == "verbose" or config_.vt_debug_level == "2") {
+    config_.vt_debug_level_val = 8;
+  } else {
+    vtAbort("Invalid value passed to --vt_deubg_level");
```
Comment (cz4rs) [inline src/vt/configs/arguments/args.cc:631]:
```suggestion
    vtAbort("Invalid value passed to --vt_debug_level");
```

File: src/vt/configs/debug/debug_print.h [hunk 2]
```diff
@@ -94,14 +94,29 @@
...
+  )
+
+#define vt_debug_print_terse(feature, ...)                              \
+  vt_debug_print_impl(                                                  \
+    false, vt::config::VTPrintConfig, terse, feature, node, __VA_ARGS__ \
+  )
+
+#define vt_debug_print_normal(feature, ...)                             \
+  vt_debug_print_impl(                                                  \
+    false, vt::config::VTPrintConfig, terse, feature, node, __VA_ARGS__ \
```
Comment (cz4rs) [inline src/vt/configs/debug/debug_print.h:119]:
```suggestion
    false, vt::config::VTPrintConfig, normal, feature, node, __VA_ARGS__ \
```

**Comment by lifflander** [comment]
There seems to be a bug in how verbose gets enabled. Might be a bug in the comparison logic.

File: src/vt/runtime/component/diagnostic_value_base.cc.tmp [hunk 3]
```diff
@@ -0,0 +1,58 @@
+/*
```
Comment (lifflander) [inline src/vt/runtime/component/diagnostic_value_base.cc.tmp:1]: This needs to be removed

**Review by cz4rs — APPROVED** [review]
Looks good to me. Please have some more eyes on the split to make sure if any debug messages should be re-categorized.

File: src/vt/runtime/component/diagnostic_value_base.cc.tmp [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact darma@sandia.gov
+//
+// *****************************************************************************
+//@HEADER
+*/
+
+#if !defined INCLUDED_VT_RUNTIME_COMPONENT_DIAGNOSTIC_VALUE_BASE_CC
```
[the remark this answers is not in the document]
Comment (cz4rs) [reply src/vt/runtime/component/diagnostic_value_base.cc.tmp:45]: Nevermind, didn't notice https://github.com/DARMA-tasking/vt/pull/1354#discussion_r605841972.

**Comment by cz4rs** [comment]
It might be worth to add a couple of prints to `hello_world` or `tutorial` example.

File: src/vt/configs/debug/debug_print.h [hunk 5]
```diff
@@ -94,14 +94,29 @@
...
+  vt_debug_print_impl(                                                  \
+    false, vt::config::VTPrintConfig, verbose, feature, node, __VA_ARGS__ \
+  )
+
+#define vt_debug_print_terse(feature, ...)                              \
+  vt_debug_print_impl(                                                  \
+    false, vt::config::VTPrintConfig, terse, feature, node, __VA_ARGS__ \
+  )
+
+#define vt_debug_print_normal(feature, ...)                             \
```
[the remark this answers is not in the document]
Comment (lifflander) [reply src/vt/configs/debug/debug_print.h:117]: There's no efficient way to overload the macro. If we do that, would would have to introduce the `meld` library I wrote, which can check for if-conditions/static checks on macros, but it's expensive so I removed it a while back. I think I would rather have `vt_debug_print` always require the category so developers can consciously think about their choice.

File: src/vt/runtime/component/diagnostic_value_base.cc.tmp (hunk 4 above)
[the remark this answers is not in the document]
Comment (lifflander) [reply src/vt/runtime/component/diagnostic_value_base.cc.tmp:45]: Removed

File: src/vt/configs/debug/debug_print.h (hunk 5 above)
[the remark this answers is not in the document]
Comment (cz4rs) [reply src/vt/configs/debug/debug_print.h:117]: :+1:

File: src/vt/configs/debug/debug_print.h (hunk 2 above)
Comment (lifflander) [reply src/vt/configs/debug/debug_print.h:119]: Fixed

File: src/vt/configs/arguments/args.cc (hunk 1 above)
Comment (lifflander) [reply src/vt/configs/arguments/args.cc:631]: Fixed

File: src/vt/configs/debug/debug_print.h [hunk 6]
```diff
@@ -165,8 +180,8 @@ struct DebugPrintOp;
 
 template <CatEnum cat, ModeEnum mod, typename Arg, typename... Args>
 static inline void debugPrintImpl(NodeType node, Arg&& arg, Args&&... args) {
-  bool const verb = vt_option_check_enabled(mod, ModeEnum::verbose);
-  if ((verb and vt::debug::preConfig()->vt_debug_verbose) or not verb) {
+  auto const level = mod & 0x000000000000000E;
```
Comment (cz4rs) [inline src/vt/configs/debug/debug_print.h:183]: This can be turned into `constexpr` (I think).

Comment (lifflander) [reply src/vt/configs/debug/debug_print.h]: I just improve this expression, but this still applies I think.

**Pull request merged**